### PR TITLE
Use Object#in? throughout the stdlib

### PR DESCRIPTION
**Pull request by Sija**
Followup to #8720

### CODE REVIEW

File: spec/std/array_spec.cr [hunk 1]
```diff
@@ -1002,7 +1002,7 @@ describe "Array" do
       [1].sample.should eq(1)
 
       x = [1, 2, 3].sample
-      [1, 2, 3].includes?(x).should be_true
+      x.in?(1, 2, 3).should be_true
```
Comment (ysbaddaden) [inline spec/std/array_spec.cr]: We are testing arrays here, I'd keep the `includes?` call. Same below.

File: src/uri/encoding.cr [hunk 2]
```diff
@@ -172,8 +172,8 @@ class URI
   # Unreserved characters are ASCII letters, ASCII digits, `_`, `.`, `-` and `~`.
   def self.unreserved?(byte) : Bool
     char = byte.unsafe_chr
-    char.ascii_alphanumeric? ||
-      {'_', '.', '-', '~'}.includes?(char)
+    char.ascii_alphanumeric? |
+      char.in?('_', '.', '-', '~')
```
Comment (ysbaddaden) [inline src/uri/encoding.cr]: bug: you changed `||` into `|`.

File: src/string.cr [hunk 3]
```diff
@@ -2260,7 +2260,7 @@ class String
   # returns the number of times the block returned a truthy value.
   #
   # ```
-  # "aabbcc".count { |c| ['a', 'b'].includes?(c) } # => 4
+  # "aabbcc".count(&.in?('a', 'b')) # => 4
```
Comment (bcardiff) [inline src/string.cr]:
```suggestion
  # "aabbcc".count &.in?('a', 'b') # => 4
```

Usually the last block is used without parenthesis in other doc examples.
There are other ocurrences also of this in the PR

Comment (Sija) [reply src/string.cr]: I'm fine with either version.

Should I remove the parenthesis from all of the examples?
- 👍 — YES
- 👎 — NO

Comment (Sija) [reply src/string.cr]: Done.

**Comment by Sija** [comment]
GTG?

**Pull request merged**